pkg/cfn/manager/api_test.go

```diff
@@ -4,20 +4,141 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/aws/aws-sdk-go-v2/service/autoscaling"
+	astypes "github.com/aws/aws-sdk-go-v2/service/autoscaling/types"
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/request"
 	"github.com/aws/aws-sdk-go/awstesting"
 	cfn "github.com/aws/aws-sdk-go/service/cloudformation"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/ginkgo/extensions/table"
 	. "github.com/onsi/gomega"
 	"github.com/stretchr/testify/mock"
 
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/cfn/builder"
 	"github.com/weaveworks/eksctl/pkg/testutils/mockprovider"
 )
 
 var _ = Describe("StackCollection", func() {
+	Context("PropagateManagedNodeGroupTagsToASG", func() {
+		It("can create propagate tag", func() {
+			// define most mock parameters
+			asgName := "asg-test-name"
```

Comment:
I was about to say extract common steps into a `BeforeEach` but seeing that nothing else is doing that in this test, I'd say meh. :D

Reply:
Ah, actually, yes! Recent change -> `GetClusterStackIfExists`. Can you please do the same? :))) For a better structure. The test looks good to me.

Reply:
@Skarlso Done in 8e5106fb5cba2f3df808523db670552af1662418
I have used `BeforeEach` to init the variables that could be take advantage of it =)

Reply:
Thanks!